Codecs/DAG-PB.md

```diff
@@ -32,24 +32,69 @@ message PBNode {
 ```
 
 The objects link names are specified in the 'Name' field of the PBLink object.
-This is non-standard and means that the 'Tsize' attribute of a link, and
-neither the 'Data' field nor the 'Links' field of the PBNode are available
-through the ipld query language. 
+All links in an object must either be blank or unique within the object.
 
-## Link Format
+## Pathing
 
-In DagCBOR, links are encoded using the raw-binary (identity, NUL) multibase in a
-field with a byte-string type (major type 2), with the tag 42. They are then
-put in the 'Hash' field of a PBLink. All links from a DAG-PB object are
-specified in the 'Links' array.
+TODO: See https://github.com/ipld/specs/issues/55
+TODO: Also see https://github.com/ipfs/unixfs-v2/issues/24 (specifically: the "Revise History" option).
+
+Neither go-ipfs nor js-ipfs agree so we have some room here because we're going to break something.
+
+### Alternative: go-ipfs
+
+In go-ipfs, we resolve use link names directly in paths: `/$name1/$name2/...`.
+Neither the Data section nor the Links section/metadata are accessible through
+paths.
+
+It's also impossible to path through nodes that don't name their links.
+
+### Alternative: js-ipfs
+
+As far as I can tell, js-ipfs supports pathing both by name and by index with
+paths like: `/Links/$name/Hash/...` or `/Links/$idx/Hash`.
+
+### Alternative: Correct IPLD
```

Comment:
Yeah, this one. Would it be difficult to implement this in the Go version? (I can't find the code) Having this unresolved and listing "Alternative"s isn't a very helpful spec. Listing what the two versions do is helpful, it'd be nice to have some concrete middle for the spec.

Reply:
These sections are just here for discussion. The final spec will have just one.

Switching to "correct IPLD" will take a bit of work in go-ipfs but we'll have to break something no matter what.